backend-config/backend-config.go

```diff
@@ -407,3 +410,167 @@ func (bc *backendConfigImpl) Identity() identity.Identifier {
+		pkgLogger.Error("Event type is not being sent for the event")
+		return false
+	}
+	eventType, isEventTypeString := eventTypeI.(string)
+	if !isEventTypeString {
+		// Seems like it makes sense!
+		pkgLogger.Errorf("Given event type :%v, cannot be casted to string", eventTypeI)
+		return false
+	}
+	isSupportedMsgType := evaluateSupportedTypes(destination.DestinationDefinition.Config, "supportedMessageTypes", eventType)
+
+	// Default behavior
+	// When something is missing in "supportedConnectionModes" or if "supportedConnectionModes" is not defined
+	// We would be checking for below things
+	// 1. Check if the event.type value is present in destination.Config["supportedMessageTypes"]
+	// 2. Check if the connectionMode of destination is cloud or hybrid(evaluated through `IsProcessorEnabled`)
+	// Only when 1 & 2 are true, we would allow the event to flow through to server
+	evaluatedDefaultBehaviour := isSupportedMsgType && destination.IsProcessorEnabled
+
+	/*
+		In future(near one) this will not be a string but rather an object
+		For every connection made to destination, we would basically have
+		```
+		{
+			connectionMode: {
+				[srcType]: oneof[cloud, hybrid, device]
+			}
+		}
+
+		```
+		Example
+		{
+			...
+			"connectionMode": {
+				"web": "", // one of cloud, hybrid or device
+				"cordova": "", // one of cloud, hybrid or device
+				"python": "", // one of cloud, hybrid or device
+				"cloud": "" // one of cloud, hybrid or device
+			}
+			...
+		}
+	*/
+	// New logic for evaluating "connectionMode"
+	var destConnModeI interface{}
+	for _, val := range []string{srcType, strings.ToLower(source.SourceDefinition.Name)} {
+		destConnModeI = misc.MapLookup(destination.Config, "connectionMode", val)
+		if destConnModeI != nil {
+			break
+		}
+	}
+	// destConnModeI := misc.MapLookup(destination.Config, "connectionMode")
+	if destConnModeI == nil {
+		return evaluatedDefaultBehaviour
+	}
+	destConnectionMode, isDestConnModeMapOfString := destConnModeI.(string)
+	if !isDestConnModeMapOfString {
+		pkgLogger.Errorf("Given destination connection mode :%v, cannot be casted to string", destConnModeI)
+		return false
+	}
+
+	supportedConnectionModesI, connModesOk := destination.DestinationDefinition.Config["supportedConnectionModes"]
+	if !connModesOk {
+		// Probably the "supportedConnectionModes" key is not present, so we rely on Default behaviour
+		return evaluatedDefaultBehaviour
+	}
+	supportedConnectionModes := supportedConnectionModesI.(map[string]interface{})
+
+	supportedEventPropsMapI := misc.MapLookup(supportedConnectionModes, destConnectionMode, srcType)
+	if supportedEventPropsMapI == nil {
+		// Most probably sourceType would not be present in "supportedConnectionModes"
+		// Might also occur when destination connection mode is not provided in "supportedConnectionModes"
+		return evaluatedDefaultBehaviour
+	}
+	supportedEventPropsMap := supportedEventPropsMapI.(map[string]interface{})
```

Comment:
what about supportedEventPropsMapI.(map[string]map[string][]string) something along these lines

Reply:
It didn't work when I tried the same thing(even before writing this piece of source-code)